ssl/quic/quic_route.c

```diff
@@ -0,0 +1,326 @@
+/*
+ * Copyright 2022 The OpenSSL Project Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License 2.0 (the "License").  You may not use
+ * this file except in compliance with the License.  You can obtain a copy
+ * in the file LICENSE in the source distribution or at
+ * https://www.openssl.org/source/license.html
+ */
+
+#include "internal/quic_route.h"
+#include "internal/list.h"
+#include "internal/thread_once.h"
+#include <openssl/lhash.h>
+#include <openssl/rand.h>
+
+struct quic_route_st {
+    OSSL_LIST_MEMBER(remote_ids, QUIC_ROUTE);
+    QUIC_ROUTE_TABLE *tbl;
+    QUIC_CONN_ID local;
+    QUIC_CONN_ID remote;
+    uint64_t     seq_no;        /* Sequence number for both ends */
+    unsigned int retired : 1;   /* Connection ID has been retired */
```

Comment:
This seems problematic. In means a route is mutable and shared between multiple threads.

Reply:
Actually, there is another way this can be done which makes the routes immutable.  Will adjust accordingly.

Reply:
Quite a bit cleaner now.